src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -1,19 +1,49 @@
 package core.basesyntax.impl;
 
 import core.basesyntax.Storage;
+import java.util.Arrays;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int MAX_ITEMS_NUMBER = 10;
+    private int size;
+    private K[] keys;
+    private V[] values;
+
+    public StorageImpl() {
+        this.keys = (K[]) new Object[MAX_ITEMS_NUMBER];
+        this.values = (V[]) new Object[MAX_ITEMS_NUMBER];
+    }
+
     @Override
     public void put(K key, V value) {
+        int index = Arrays.asList(keys).indexOf(key);
+        if (index != -1) { //если такой key есть в массиве keys
+            this.values[index] = value; //то перезаписываем его value
+            if (index >= size) {
+                this.size++;
+            }
+        } else { //если нет
+            this.keys[this.size] = key; //то добавляем новую запись key & value
+            this.values[this.size] = value;
+            this.size++;
+        }
```

Comment:
```suggestion
        if (index != -1) { //если такой key есть в массиве keys
            values[index] = value; //то перезаписываем его value
            if (index >= size) {
                this.size++;
            }
        } else { //если нет
            keys[this.size] = key; //то добавляем новую запись key & value
            values[this.size] = value;
            size++;
        }
```

Давай уберем отовсюду ключевое слово `this`, оно нам тут не нужно(у нас нету глобальных и локальных прееменных с одинаковым названием)